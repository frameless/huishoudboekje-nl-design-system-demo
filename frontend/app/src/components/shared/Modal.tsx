import {Modal as ChakraModal, ModalBody, ModalCloseButton, ModalContent, ModalFooter, ModalHeader, ModalOverlay, ModalProps as ChakraModalProps} from "@chakra-ui/react";
import React from "react";

type ModalProps = Omit<ChakraModalProps, "isOpen"> & {
	title: string,
}

const Modal: React.FC<ModalProps> = ({
	title,
	children,
	...props
}) => {
	return (
		<ChakraModal isOpen={true} {...props}>
			<ModalOverlay />
			<ModalContent>
				<ModalHeader>{title}</ModalHeader>
				<ModalCloseButton />
				<ModalBody>{children}</ModalBody>
				<ModalFooter />
			</ModalContent>
		</ChakraModal>
	);
};

export default Modal;