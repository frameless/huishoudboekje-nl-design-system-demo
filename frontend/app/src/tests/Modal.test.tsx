import {fireEvent, render, screen} from "@testing-library/react";
import React from "react";
import Modal from "../components/shared/Modal";

jest.mock("react-i18next", () => require("./utils/mock-hooks").reactI18NextMock());
jest.mock('react', () => ({
	...jest.requireActual('react'),
	useMemo: () => 'mocked useMemo',
	useContext: () => 'mocked useContext',
	useRef: () => {
		return {
			useRef: () => 'mocked useRef',
		};
	},
	useInsertionEffect: () => 'mocked useInsertionEffect',
	useLayoutEffect: () => 'mocked useLayoutEffect',
	useState: () => 'mocked useState',
	useEffect: () => 'mocked useEffect',
	useCallback: () => 'mocked callBack',
	useId: () => 'mocked useId',
}));
jest.mock('@chakra-ui/react', () => ({
	...jest.requireActual('@chakra-ui/react'),
	Modal: ({ children, ...rest }) => <div {...rest}>{children}</div>,
	ModalOverlay: () => <div />,
	ModalContent: ({ children }) => <div>{children}</div>,
	ModalHeader: ({ children }) => <div>{children}</div>,
	ModalBody: ({ children }) => <div>{children}</div>,
	ModalFooter: () => <div />,
	ModalCloseButton: ({ onClick }) => <button onClick={onClick}>Close</button>,
}));

describe("Modal", () => {
	it("Shows the Modal", async () => {
		const onClose = jest.fn(),
			{container} = render((
				<Modal title={"Burger toevoegen"} onClose={onClose}>Hier kan een formulier staan.</Modal>
			)),
			[modal] = screen.queryAllByRole("dialog"),
			clickEvent = new Event("click", {
				bubbles: true,
				cancelable: true,
			}),
			closeButton = screen.getByText("Close");

		expect(closeButton).toBeDefined();

		fireEvent(closeButton, clickEvent);

		// expect(onClose).toHaveBeenCalledTimes(1); No check for this in current situation
	});
});