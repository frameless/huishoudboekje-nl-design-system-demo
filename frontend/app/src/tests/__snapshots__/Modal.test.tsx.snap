// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Shows the Modal 1`] = `
<section
  aria-describedby="chakra-modal--body-:r0:"
  aria-labelledby="chakra-modal--header-:r0:"
  aria-modal="true"
  class="chakra-modal__content css-6yiygt"
  id="chakra-modal-:r0:"
  role="dialog"
  style="opacity: 0; transform: scale(0.95) translateZ(0);"
  tabindex="-1"
>
  <header
    class="chakra-modal__header css-72fd9l"
    id="chakra-modal--header-:r0:"
  >
    Burger toevoegen
  </header>
  <button
    aria-label="Close"
    class="chakra-modal__close-btn css-6sdaxy"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-onkibi"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
        fill="currentColor"
      />
    </svg>
  </button>
  <div
    class="chakra-modal__body css-0"
    id="chakra-modal--body-:r0:"
  >
    Hier kan een formulier staan
  </div>
  <footer
    class="chakra-modal__footer css-192qrng"
  />
</section>
`;
